Update to pegasus 5.0.3

spxiwh opened the pull request:
Pegasus 5.0.3 has been released which contains a few important fixes for PyCBC:

* It is now possible to have a condorpool_symlink and condorpool_copy site, and they'll now work (with some tweaks to the site catalog to add new options).
* We now have the setup needed for OSG running. Note that we are waiting on a condor fix (currently slated for 10.1, but we've asked for it to go in 10.0), which can cause failures when copying files *back* from OSG if the file LFN includes a relative path (e.g. 140567/H1-INSPIRAL-BLAH.hdf).  However, for use-cases that don't use this (GRB, for e.g. probably doesn't want to use GPS time prefixes) we should be fully ready to use OSG "easily".

I am also requesting this be updated on the LDG machines, I suggest folks do so on local clusters as well because you will need a `pegasus-plan` from the 5.0.3 release to work with these changes.

ahnitz commented on requirements.txt:
```diff
@@ -17,7 +17,8 @@ tqdm
 # Requirements for full pegasus env
-pegasus-wms >= 5.0.1
+pegasus-wms.api >= 5.0.3
+GitPython>1.0 # Required by pegasus-wms.api, but not declared as such
```
Should we flag this upstream?

spxiwh replied:
Possibly ... First want to check if this actually is the case. I'm not sure if this is something like GitPython should have been installed alongside the main pegasus executables at the system level, but instead it can find it here.

ahnitz replied:
Ah, ok. Have you checked this is really needed for generating the workflow though? It used to be the case that GitPython was just needed for some of the pegasus exes but not the actual API. If that is still the case, then a CI issue might be different as GitPython should have been pulled in by the pegasus system install.

spxiwh replied:
When I didn't have this the pegasus-plan jobs were failing. So I think this probably isn't a requirement of pegasus-wms.api, but a requirement of pegasus-plan that isn't being pulled in. Maybe the right approach is to install GitPython alongside the pegasus install in the github workflows.

ahnitz reviewed (commented):
Ok, well, except for the commented issue on where the dependency should really be, I think this looks OK. Once that issue is resolved and the tests pass, I am happy for this to merge.

spxiwh commented:
@ahnitz I've now moved the GitPython requirement into the github workflow .yml scripts. I think we weren't installing pegasus correctly there, and I don't think it was pulling its dependencies. I fixed this by matching to how we install condor in these scripts, and verified that this was pulling the "python3-git" package that pegasus declares.

However, pegasus-plan still doesn't find it, and I don't know why. So I had to add a final patch to explicitly pip install GitPython. I expect this will now work, but don't understand why it didn't work before the final patch!
